### PR TITLE
use transitive_proto_path_flags if exist

### PR DESCRIPTION
**Pull request by ittaiz**
following https://github.com/bazelbuild/bazel/issues/4544 I'm adding an optional propagation of transitive_proto_path_flags to protoc.
It's optional since it's not currently in a released version and in fact the exposition to skylark will only land in an hour or two but I've tested it manually.
For the above reason the test is under `test_expect_failure` and not part of CI since we can't run only that test against bazel head.
@azymnis @johnynek I'd like to merge this soon so I'd appreciate your review

### CODE REVIEW

**Comment by azymnis** [comment]
@ittaiz LGTM

File: scala_proto/scala_proto.bzl [hunk 1]
```diff
@@ -320,12 +320,15 @@ def _colon_paths(data):
...
 def _gen_proto_srcjar_impl(ctx):
     acc_imports = depset()
+    transitive_proto_path_flags = depset()
 
     proto_deps, jvm_deps = [], []
     for target in ctx.attr.deps:
         if hasattr(target, 'proto'):
             proto_deps.append(target)
             acc_imports += target.proto.transitive_sources
+            if hasattr(target.proto, 'transitive_proto_path_flags'):
```
Comment (johnynek) [inline scala_proto/scala_proto.bzl]: we should have either an issue or a code comment on these version compatibility check things so we don't lose track.

Like after version `0.11` is the oldest supported version, remove this check or something. What do you think?

I'm worried about the code building up layers of these checks we don't recall why the exist.

**Comment by ittaiz** [comment]
@johnynek added the comment, wdyt?

**Comment by ittaiz** [comment]
I’ll wait at least for the ci.bazel.build one to pass
One caveat- @iirina hasn’t committed the change to bazel that exposes transitive_proto_path_flags to skylark. If for whatever reason she decides to use a different name I’ll need to do another PR but I highly doubt there will be a problem since I came up with name according to the convention

**Comment by ittaiz** [comment]
Merging since I think ci is stuck. The additional commit was just comment addition

**Pull request merged**